### PR TITLE
Make BulkWriter private until protos are released

### PR DESCRIPTION
**Pull request by thebrianchen**
- Remove BulkWriter mentions from docs.
- Remove export for `BulkWriter and `BulkWriterOptions` in `firestore.d.ts`.
- Prefix `bulkWriter()` with `_` in `index.ts`

### CODE REVIEW

File: types/firestore.d.ts [hunk 1]
```diff
@@ -416,7 +416,7 @@ declare namespace FirebaseFirestore {
    *
    * @class
    */
-  export class BulkWriter {
+  class BulkWriter {
```
Comment (schmidt-sebastian) [inline types/firestore.d.ts:419]: All types in a `d.ts` file as exported by default. Unfortunately, we can only hide them by removing the entries completely.

File: dev/src/index.ts [hunk 2]
```diff
@@ -677,7 +678,7 @@ export class Firestore {
    * });
    * bulkWriter.close();
    */
-  bulkWriter(options?: BulkWriterOptions): BulkWriter {
+  _bulkWriter(options?: BulkWriterOptions): BulkWriter {
```
Comment (schmidt-sebastian) [inline dev/src/index.ts:681]: Make sure to also remove the `export { BulkWriter }` line.

File: types/firestore.d.ts (hunk 1 above)
Comment (thebrianchen) [reply types/firestore.d.ts:419]: done.

File: dev/src/index.ts (hunk 2 above)
Comment (thebrianchen) [reply dev/src/index.ts:681]: done.

**Pull request merged**